synapse/storage/_base.py

```diff
@@ -515,6 +558,7 @@ def _simple_upsert(self, table, keyvalues, values,
             insertion_values (dict): additional key/values to use only when
                 inserting
             lock (bool): True to lock the table when doing the upsert.
+            best_effort (bool): If we run into a transaction error, do we stop trying?
```

Comment:
Can you also document the fact that we return `None` if we use a native upsert?

Reply:
done